### PR TITLE
Add --bundlerConfig to cli commands so you can build a nested dir w/o changing --cwd.

### PR DESCRIPTION
**Pull request by mikeparisstuff**
# The problem

**Issue: https://github.com/sveltejs/sapper/issues/1074**

The launcher script created in `cli.ts` is placed in a different `[dest]` path than the rest of the build when you provide both a `--cwd` and `[dest]` in the command. This change causes the launcher file to also take the `--cwd` into account and places the launcher file in the correct location.

## Example

If you have a sapper project at `/Users/me/projects/sapper_app` and you run this command: 

```
sapper build ./sapper_out --cwd ./sapper_app
```

from `/Users/me/projects` then you get the following build output

```
/Users/me/projects/sapper_out
   /index.js
/Users/me/projects/sapper_app/sapper_out
   /build.json
   /template.html
   /service-worker.js
   /client/
   /server/
```

 This PR fixes a few issues with the `--cwd` but does not fix all of them. I switched gears midway through to solve the problem with a new argument `--bundlerConfig`, but I'm pretty sure the fixes to `--cwd` are still in the right direction so I left them in place for now.

This PR makes three changes:

1. The fix is a simple change to include the `--cwd` in the launcher path, if one exists, and places `index.js` in the same directory as the rest of the build. This was not enough to fix the problem but I believe is still a bug fix.
2. Adds a new `--bundlerConfig` arg allows you to point to the `rollup.config.js` or `webpack.config.js` which is still expected to be in the root relative to the `--cwd` and is the last missing arg outside of `--src`, `--output`, `--static`, and `--routes`. 
3. Changes the logic in the generated `server.js` to find `build.json` based on a path relative to `server.js` instead of relative to the `--cwd` at build time. This allows you to build a sapper app in one place and then run it in another.

Changes 2 & 3 allow you to run:

```
sapper build __sapper__ --src client/sapper/src --routes client/sapper/src/routes --static client/sapper/static --output client/sapper/node_modules/@sapper --bundlerConfig client/sapper/rollup.config.js --bundler rollup
```

which results in a build that can be packaged, moved, and run in the same way. This means you can run the app using `node __sapper__` or, if the directory is somewhere else, `node some/nested/path/__sapper__`.

### Tests

The tests pass and I have a successful build & run working now 👍

### CODE REVIEW

**Comment by mikeparisstuff** [comment]
After fixing `build --cwd [dest]` in the build, I was trying `dev --cwd sapper_app --build-dir ../sapper_dev` and it failed with error:

```
> ENOENT: no such file or directory, open 'src/template.html'
```

The issue was very similar and was that the `src` variable in the constructor in `dev.ts` was being used in one location without taking `--cwd` into account and another where it was.

I went ahead and added the fix to this PR and have confirmed that `dev --cwd` now works.

**Comment by mikeparisstuff** [comment]
The original fixes in this CR that tried to fully support `--cwd` were not enough to solve the problem. I shifted gears and instead decided to try to use `--src`, `--routes`, `--static`, and `--output` to change the source locations without changing the `--cwd`. To make this work, I had to change how the `rollup.config.js` and `webpack.config.js` as those were expected to be in the root of the `--cwd`. I added a `--bundlerConfig` option to build, dev, and export so that you can specify where the bundler config file should be found. This change allowed me to get a working build going.

After the --bundlerConfig was added, I went and changed how the `build.json` file was found in the generated `server.js`. This was trying to use a hardcoded path relative to the `--cwd` at build time which introduced some issues in sandboxed build environments like bazel where paths are not easily assumed. The `build.json` is always in the same spot relative to the `server.js` file so I changed it to look relative to that file. After these two changes, I am able to build the project and run it successfully. Success :)

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "sapper",
+  "name": "motley-sapper",
```
Comment (benmccann) [inline package.json:2]: this shouldn't change

**Comment by pngwn** [comment]
API additions have to meet a pretty high bar in general but we won't be looking at this until some higher-level questions about how we handle configuration have been answered.

**Comment by ajbouh** [comment]
I've applied this PR to my own build and discovered that switching to `__dirname` in the generated code requires an additional bit of config if using the webpack bundler.

To prevent webpack from hardcoding `__dirname` to `"/"`, you need to modify your webpack.config.js to look something like:

```
module.exports = {
  // ...
  server: {
    // ...
    node: {
      __dirname: false
    }
  }
}
```

**Comment by benmccann** [comment]
Thanks for putting together this PR! Sapper never had a good method of handling configuration, which made it difficult to know how to integrate this feature. However, that's been addressed in SvelteKit by adding a `svelte.config.cjs` file and adapters. I'm going to go ahead and close this PR since we won't be adding the feature to Sapper. You might like to check out SvelteKit instead. We'll be reviewing PRs for missing features there

**Pull request closed without merging**